WooCommerce/Classes/ViewRelated/ReusableViews/AuthWebView/AuthenticationService.swift

```diff
@@ -0,0 +1,221 @@
+
+        cookieJar.hasWordPressComAuthCookie(
+            username: username,
+            atomicSite: false) { hasCookie in
+
+                guard !hasCookie else {
+                    // The stored cookie can be stale but we'll try to use it and refresh it if the request fails.
+                    success()
+                    return
+                }
+
+                self.getAuthCookiesForWPCom(username: username, authToken: authToken, success: { cookies in
+                    cookieJar.setCookies(cookies) {
+
+                        cookieJar.hasWordPressComAuthCookie(username: username, atomicSite: false) { hasCookie in
+                            guard hasCookie else {
+                                failure(RequestAuthCookieError.wpcomCookieNotReturned)
+                                return
+                            }
+                            success()
+                        }
+
+                    }
+                }) { error in
+                    // Make sure this error scenario isn't silently ignored.
+                    ServiceLocator.crashLogging.logError(error)
+
+                    // Even if getting the auth cookies fail, we'll still try to load the URL
+                    // so that the user sees a reasonable error situation on screen.
+                    // We could opt to create a special screen but for now I'd rather users report
+                    // the issue when it happens.
+                    failure(error)
+                }
+        }
+    }
+
+    func getAuthCookiesForWPCom(
+        username: String,
+        authToken: String,
+        success: @escaping (_ cookies: [HTTPCookie]) -> Void,
+        failure: @escaping (Error) -> Void) {
+
+        let loginURL = URL(string: AuthenticationService.wpComLoginEndpoint)!
+        let headers = [
+            "Authorization": "Bearer \(authToken)"
+        ]
+        let parameters = [
+            "log": username,
+            "rememberme": "true"
+        ]
+
+        requestAuthCookies(
+            from: loginURL,
+            headers: headers,
+            parameters: parameters,
+            success: success,
+            failure: failure)
+    }
+
+    // MARK: - Request Construction
+
+    private func requestAuthCookies(
+        from url: URL,
+        headers: [String: String],
+        parameters: [String: String],
+        success: @escaping (_ cookies: [HTTPCookie]) -> Void,
+        failure: @escaping (Error) -> Void) {
+
+        // We don't want these cookies persisted in other sessions
+        let session = URLSession(configuration: .ephemeral)
+        var request = URLRequest(url: url)
+
+        request.httpMethod = "POST"
+        request.httpBody = body(withParameters: parameters)
+
+        headers.forEach { (key, value) in
+            request.setValue(value, forHTTPHeaderField: key)
+        }
+//        request.setValue(WPUserAgent.wordPress(), forHTTPHeaderField: "User-Agent")
```

Comment:
Do we need to set up another user agent here?

Reply:
I don't see a reason for that. Nothing changed when I copy-pasted exact user-agent string from WP.
Without explicit setter it will be same as default: `WooCommerce/10.8.0.0 CFNetwork/1390 Darwin/22.1.0`.